renpy/ast.py

```diff
@@ -2521,19 +2521,23 @@ def set_default(self, start):
             d["_defaults_set"] = defaults_set = renpy.revertable.RevertableSet()
             d.ever_been_changed.add("_defaults_set")
 
-        if self.varname not in defaults_set:
+        if self.varname in defaults_set:
+            if start and renpy.config.developer:
+                raise Exception("{}.{} is being given a default a second time.".format(self.store, self.varname))
+            return
 
-            if start or (self.varname not in d.ever_been_changed):
-                d[self.varname] = renpy.python.py_eval_bytecode(self.code.bytecode)
+        # do the variable shadowing if not in this case, for compatibility reasons
+        if start and (renpy.config.developer is True):
+            fullname = '.'.join((self.store, self.varname))
+            if fullname in renpy.python.store_dicts.keys():
+                raise Exception("The {} default variable is shadowing a store with the same name.".format(fullname))
```

Comment:
```suggestion
                raise Exception("{} is being given a default but is already a store.".format(fullname))
```
Some creators won't understand what shadowing is in this context, so copying the style of the existing exception and explaining that it's already a store should hopefully be more immediately understandable to them.